File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
# M개 중 N개 뽑는 경우의 수 mCn과 같음
T = int(input())
for _ in range(T):
    n, m = map(int, input().split())
    
    result = 1
    if n!=m:
        if m-n>m//2:
            for i in range(m, m-n, -1):
                result*=i
            for j in range(n, 0, -1):
                result//=j
        else: # 8C6
            for i in range(m, n, -1):
                result*=i
            for j in range(m-n, 0, -1):
                result//=j

    print(int(result))

File: 백준/Silver/1010. 다리 놓기/다리 놓기.cc
#include <bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    
    int t;
    cin >> t;
    
    for (int i=0;i<t;i++){
        int n, m;
        cin >> n >> m;
        long long result = 1;
        //mCn
        if (m!=n){ // 8C6 = 8C2
            if (n > m-n){
                n = m-n;
            }
            for (int j=0;j<n;j++){
                result = result * (m-j) / (1+j);
            }
        }
        cout << result << "\n";
    }
    
    return 0;
}